transport.go

```diff
@@ -32,27 +32,73 @@ var quicConfig = &quic.Config{
 }
 
 type connManager struct {
-	connIPv4Once sync.Once
-	connIPv4     net.PacketConn
+	mu sync.Mutex
+	// underhood PacketConn to its (re)use count
+	ipv4Conns map[net.PacketConn]int
+	ipv6Conns map[net.PacketConn]int
+}
 
-	connIPv6Once sync.Once
-	connIPv6     net.PacketConn
+func newConnManager() *connManager {
+	return &connManager{
+		ipv4Conns: make(map[net.PacketConn]int),
+		ipv6Conns: make(map[net.PacketConn]int),
+	}
 }
 
-func (c *connManager) GetConnForAddr(network string) (net.PacketConn, error) {
+func pickDialConn(conns map[net.PacketConn]int, remoteHostIP net.IP) net.PacketConn {
+	backup := make([]net.PacketConn, 0, len(conns))
```

Comment:
So, this is going to be called *a lot* (every dial). Let's just set *one* backup if and only if a backup hasn't been set. We don't really need to collect a list of them and then select the first.

Reply:
Nice, will do.